File: 11-container-with-most-water/11-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int begin = 0;
        int end = height.length-1;
        int maxiarea = 0;
        
        while(begin<end){
            int height1 = height[begin];
            int height2 = height[end];
            
            int miniheight = Math.min(height1,height2);
            
            int temparea = miniheight*(-begin+end);
            if(maxiarea<temparea) maxiarea = temparea;
            
            if(height1<height2) begin++;
            else end--;
        }
        return maxiarea;
    }
}

File: 11-container-with-most-water/11-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int lo = 0;
        int hi = height.size()-1;
        int mxarea = 0;
        while(lo<hi){
            int temp = min(height[lo],height[hi]);
            int diff = hi-lo;
            if(diff==0)diff++;
            int temparea = temp*diff;
            mxarea = max(mxarea,temparea);
            if(height[lo]>height[hi])hi--;
            else lo++;
        }
        return mxarea;
    }
};